pkcs7/src/cryptographic_message_syntax2004.rs

```diff
@@ -0,0 +1,378 @@
+     kekid KEKIdentifier,
+     keyEncryptionAlgorithm KeyEncryptionAlgorithmIdentifier,
+     encryptedKey EncryptedKey }
+
+   KEKIdentifier ::= SEQUENCE {
+     keyIdentifier OCTET STRING,
+     date GeneralizedTime OPTIONAL,
+     other OtherKeyAttribute OPTIONAL }
+
+   PasswordRecipientInfo ::= SEQUENCE {
+     version CMSVersion,   -- always set to 0
+     keyDerivationAlgorithm \[0\] KeyDerivationAlgorithmIdentifier
+                                OPTIONAL,
+     keyEncryptionAlgorithm KeyEncryptionAlgorithmIdentifier,
+     encryptedKey EncryptedKey }
+
+   OtherRecipientInfo ::= SEQUENCE {
+     oriType OBJECT IDENTIFIER,
+     oriValue ANY DEFINED BY oriType }
+
+   DigestedData ::= SEQUENCE {
+     version CMSVersion,
+     digestAlgorithm DigestAlgorithmIdentifier,
+     encapContentInfo EncapsulatedContentInfo,
+     digest Digest }
+
+   Digest ::= OCTET STRING
+
+   EncryptedData ::= SEQUENCE {
+     version CMSVersion,
+     encryptedContentInfo EncryptedContentInfo,
+     unprotectedAttrs \[1\] IMPLICIT UnprotectedAttributes OPTIONAL }
+
+   AuthenticatedData ::= SEQUENCE {
+     version CMSVersion,
+     originatorInfo \[0\] IMPLICIT OriginatorInfo OPTIONAL,
+     recipientInfos RecipientInfos,
+     macAlgorithm MessageAuthenticationCodeAlgorithm,
+     digestAlgorithm \[1\] DigestAlgorithmIdentifier OPTIONAL,
+     encapContentInfo EncapsulatedContentInfo,
+     authAttrs \[2\] IMPLICIT AuthAttributes OPTIONAL,
+     mac MessageAuthenticationCode,
+     unauthAttrs \[3\] IMPLICIT UnauthAttributes OPTIONAL }
+
+   AuthAttributes ::= SET SIZE (1..MAX) OF Attribute
+
+   UnauthAttributes ::= SET SIZE (1..MAX) OF Attribute
+
+   MessageAuthenticationCode ::= OCTET STRING
+
+   DigestAlgorithmIdentifier ::= AlgorithmIdentifier
+
+   SignatureAlgorithmIdentifier ::= AlgorithmIdentifier
+
+   KeyEncryptionAlgorithmIdentifier ::= AlgorithmIdentifier
+
+   ContentEncryptionAlgorithmIdentifier ::= AlgorithmIdentifier
+
+   MessageAuthenticationCodeAlgorithm ::= AlgorithmIdentifier
+
+   KeyDerivationAlgorithmIdentifier ::= AlgorithmIdentifier
+
+   RevocationInfoChoices ::= SET OF RevocationInfoChoice
+
+   RevocationInfoChoice ::= CHOICE {
+     crl CertificateList,
+     other \[1\] IMPLICIT OtherRevocationInfoFormat }
+
+   OtherRevocationInfoFormat ::= SEQUENCE {
+     otherRevInfoFormat OBJECT IDENTIFIER,
+     otherRevInfo ANY DEFINED BY otherRevInfoFormat }
+
+   CertificateChoices ::= CHOICE {
+     certificate Certificate,
+     extendedCertificate \[0\] IMPLICIT ExtendedCertificate,  -- Obsolete
+     v1AttrCert \[1\] IMPLICIT AttributeCertificateV1,        -- Obsolete
+     v2AttrCert \[2\] IMPLICIT AttributeCertificateV2,
+     other \[3\] IMPLICIT OtherCertificateFormat }
+
+   AttributeCertificateV2 ::= AttributeCertificate
+
+   OtherCertificateFormat ::= SEQUENCE {
+     otherCertFormat OBJECT IDENTIFIER,
+     otherCert ANY DEFINED BY otherCertFormat }
+
+   CertificateSet ::= SET OF CertificateChoices
+*/
```

Comment:
Please don't leave large comment blocks in the code.

Reply:
These were left specifically because this incomplete and the comment highlights what has not been implemented. If preference is to drop, that's fine with me.